src/core/wallet/Wallet.ts

```diff
@@ -195,6 +212,227 @@ class Wallet extends Channel<WalletEventName> {
             (a, b) => (a && b ? a !== b : true),
         )
     }
+
+    protected async getWalletDataAll() {
+        logger("getWalletDataAll")
+
+        const account = this.account
+        const contracts = this._perp.contracts
+        const multicall2 = new MulticallReader({ contract: contracts.multicall2 })
+
+        const callsMap: Record<string, ContractCall[]> = {}
+        this._collateralTokenList.forEach(collateralToken => {
+            const isSettlementToken = collateralToken instanceof SettlementToken
+            const collateralTokenAddress = collateralToken.address
+            const collateralTokenContract = collateralToken.contract
+            const decimalsCall = isSettlementToken
```

Comment:
nit: too many ternary operator makes code hard to read. How about just extract `settlement token` from collateral token list and do it in order instead of do condition statement everytime

Reply:
Agree. Let's add a note first, then optimize it in another PR.